### ISSUE TITLE
Dark trails in LBL CCDs

### ISSUE THREAD

**Issue by julienguy** [desihub/desispec#796]
Opening an issue for the record for a problem this is already (partly) solved:

Some LBL CCDs exhibit negative (dark) trails along CCD rows. This is a CCD/readout issue, it has nothing to do with the optics. Preliminary tests have shown it might be difficult to fix this in the readout.

It is a small effect ~ 0.1% that can be fixed when preprocessing the images.
A correction of the following form (for amplifiers B and D) works well:
```
image[j,i] += sum_{k<i} image[j,k] * a * |i-k|/b * exp( -|i-k|/b)
```
with i = CCD column index ;  j = CCD row index.

Example for R5 a=0.0012 b=6 pixels.

The following image is showing the effect the correction.
![r5-darktrail-correction](https://user-images.githubusercontent.com/5192160/61414831-d51b9100-a8a3-11e9-9f25-5f258a71a909.png)

**Comment by djschlegel** [comment]
This won't work well when the A/D is saturated at 65536 where the information is lost and this will underestimate the effect for those counts above saturation.  But maybe that's a rare enough occurrence in real data.

**Comment by sbailey** [comment]
Visually that is quite impressive for how well it works.  I'm a bit surprised by the functional form though which has the correction peaking at 6 pixels away instead of being just an exponential dropoff (or similar).  i.e. how sure are you that there isn't a correction "inside" an individual PSF?

Since this correction peaks in the middle of a neighboring trace (with a fully packed slit), we could consider masking pixels whose correction is larger than X% of the value in that pixels.  e.g. if a bright star is next to a faint source and we're making a 50% correction to some pixel due to it's neighboring fiber, we might prefer to just mask it rather than treat it as good.  For consideration (and further study).

**Comment by julienguy** [comment]
There must be a correction to make inside the PSF but we cannot disentangle it from the PSF shape itself, and it is not a problem because we can absorb it in the PSF modeling.

**Comment by julienguy** [comment]
The code to fit the trail and do the correction during preprocessing is implemented in the branch darktrail.
The results for the 4 amplifiers which have this problem are shown below.
r0-B width=13 amp=0.00024
![r0-darktrail-correction](https://user-images.githubusercontent.com/5192160/61552580-10859f00-aa0d-11e9-8d3c-4fdb9e88f401.png)
r1-D width=18 amp=0.00021
![r1-darktrail-correction](https://user-images.githubusercontent.com/5192160/61552608-21361500-aa0d-11e9-8622-39d868a54b3b.png)
r2-B width=23 amp=0.0002
![r2-darktrail-correction](https://user-images.githubusercontent.com/5192160/61552628-2c894080-aa0d-11e9-9f29-0b3e0c281741.png)
r5-B width=6 amp=0.0012
![r5-darktrail-correction](https://user-images.githubusercontent.com/5192160/61552666-4591f180-aa0d-11e9-8d44-6c7435671d9a.png)

**Comment by sbailey** [comment]
Fixed in PR #798.  Closing for now, while noting dangling questions if we have to come back to this:
* If the correction to a neighboring fiber is large compared to the flux in that fiber, do we want to set a mask bit?
* Do we need to handle saturation differently?  e.g. maybe we don't mask a normal correction, but if some correction based upon a saturated pixel is large compared to the flux it is correcting, then it gets masked.
* Are any asymmetries from incorrect functional form or parameters sufficiently absorbed by the PSF model? (current feeling is "yes", but this might come up again)
